department-of-veterans-affairs/content-build
# [Medallia] Add functionality for subpaths in Medallia

## Summary

This ticket updates the liquid filter 'getSurvey' to recognize subpaths of a URL if a custom Medallia survey exists for that URL.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#75521

## Testing done

Medallia cannot launched via local build due to permissions, so it must be tested on the review instance.

-The `getSurvey` liquid filter previously required subpaths to be explicitly defined in the Medallia survey JSON object.
- All unit tests were updated 

Please follow the steps below:

Test returning custom survey with subpaths

1. Go to the page `/health-care/eligibility`.
2. Click on the FEEDBACK button.
3. The modal for survey `41` should appear.

Test returning custom survey number without subpaths

1. Go to the page `/search`.
2. Click on the FEEDBACK button.
3. The modal for survey `20` should appear.

Test returning default survey:

1. Go to the homepage `/`.
2. Click on the FEEDBACK button.
3. The modal for survey `11` should appear.

## What areas of the site does it impact?
Any page with a Medallia FEEDBACK button at the bottom of the main content


## Acceptance criteria

- [x] Given a valid URL that matches an entry in either surveyData.urls or surveyData.urlsWithSubPaths, then it should return the correct survey ID associated with the URL
- [x] Given a URL that does not directly match an entry in surveyData.urls but is a subpath of an entry in surveyData.urlsWithSubPaths, then it should return the correct survey ID associated with the URL
- [x] Given a URL that does not match any entry in surveyData.urls or any subpath in surveyData.urlsWithSubPaths, then it should return a default survey ID

### Quality Assurance & Testing
